### PR TITLE
Unify usage of dashboard title

### LINKED ISSUE #18850: New dashboards do not have the title of dashboard in config
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When creating a new dashboard, you can configure a title, icon and URL. 

CleanShot 2023-12-01 at 00 09 19

When I look at the generated config, the title is not stored in the dashboard config. If you started from scratch, it contains the following:

```yaml
views:
  - title: Home
```

### Describe the behavior you expected

I would expect the title of the dashboard to be stored in the config, since I just gave it a name. If I set the title to <my dashboard>, I would expect to see that.

```
title: <my dashboard>
views:
  - title: Home
```

### Steps to reproduce the issue

1. Go to dashboard config
2. Add new dashboard
3. Start from scratch
4. Give it a name and click create
5. Go to dashboard -> the toolbar is empty.
6. Edit config -> no title set

### What version of Home Assistant Core has the issue?

core-2023.11.4

### What was the last working version of Home Assistant Core?

Don't know

### In which browser are you experiencing the issue with?

Arc

### Which operating system are you using to run this browser?

MacOS

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by piitaya**
## Breaking changes

`title` in yaml is no longer used. It was used for cast and when only one view was display. It will now use the panel title.

## Proposed change

Uses only one title for dashboard.
Previously, we had 2 titles for dashboard : one defined per panel (dashboard config) and one defined in lovelace config.
This PR removes the lovelace config one to only use the panel title.

The edit button at the top now open the edit dashboard dialog.

Cast and YAML dashboards have also be tested with these changes.

CleanShot 2024-05-22 at 18 19 29

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes https://github.com/home-assistant/frontend/issues/18850
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
  - Enhanced dashboard management with new functions for creating, editing, and deleting dashboards.
  - Improved title retrieval based on URL paths for more dynamic and context-aware dashboard titles.

- **Bug Fixes**
  - Adjusted rendering logic and styling for item icons to ensure consistent display across different components.

- **Refactor**
  - Simplified methods and removed unnecessary parameters in dashboard strategy classes.

- **Chores**
  - Updated import statements and method calls to use new functions for panel title retrieval.

These changes enhance the overall user experience by providing more flexible and intuitive dashboard management capabilities.

### CODE REVIEW

File: src/panels/lovelace/ha-panel-lovelace.ts [hunk 1]
```diff
@@ -120,6 +120,7 @@ export class LovelacePanel extends LitElement {
     if (panelState === "loaded") {
       return html`
         <hui-root
+          .panel=${this.panel}
```
[the remark this answers is not in the document]
Comment (piitaya) [reply src/panels/lovelace/ha-panel-lovelace.ts:123]: This can be undefined

[the remark this answers is not in the document]
Comment (piitaya) [reply src/panels/lovelace/ha-panel-lovelace.ts:123]: Yes do it.

[the remark this answers is not in the document]
Comment (piitaya) [reply src/panels/lovelace/ha-panel-lovelace.ts:123]: No

File: src/panels/lovelace/hui-root.ts [hunk 2]
```diff
@@ -758,8 +772,41 @@ class HUIRoot extends LitElement {
...
+            confirmText: this.hass!.localize("ui.common.delete"),
+            destructive: true,
+          }))
+        ) {
+          return false;
+        }
+        await deleteDashboard(this.hass!, dashboard!.id);
+        return true;
+      },
+      createDashboard: async () => {
```
Comment (bramkragten) [inline src/panels/lovelace/hui-root.ts:806]: Can we make this param optional?

**Review by bramkragten — APPROVED** [review]

**Pull request merged**